zkevm-circuits/src/bytecode_circuit/circuit.rs

```diff
@@ -38,10 +75,8 @@ pub struct BytecodeCircuitConfig<F> {
     value_rlc: Column<Advice>,
     length: Column<Advice>,
     push_data_size: Column<Advice>,
-    push_data_left_inv: Column<Advice>,
-    push_data_left_is_zero: IsZeroConfig<F>,
-    index_length_diff_inv: Column<Advice>,
-    index_length_diff_is_zero: IsZeroConfig<F>,
+    push_data_left_is_zero: IsZeroChip<F>,
```

Comment:
I did a quick search of the codebase. 

- StateCircuitConfig has `BatchedIsZeroConfig` defined in the struct and `BatchedIsZeroChip` wrapping the config everywhere.
- PiCircuitConfig has `IsZeroChip`.

I'd argue using chips is better for `IsZeroChip` case because it has the automatic inverse assignment method implemented.

I'd also argue it is redundant for both IsZero and BatchIsZero gadgets to have both XConfig and XChip structs. The rationale for having config and chip separated is to implement the Chip trait. But I don't see we're getting any benefit in our use cases.

- For the `IsZero` gadget. The `Chip` trait did nothing really interesting.
- For the `BatchIsZero` gadget. We don't even implement `Chip` trait for it.

Reply:
Let's continue the discussion in #1401